crates/rover-client/src/introspection/schema.rs

```diff
@@ -0,0 +1,295 @@
+                                        let list1 = FieldType::new_list(ty, true);
+                                        let list2 = FieldType::new_list(list1, false);
+                                        let mut field_def = Field::new(field.name, list2);
+                                        field_def.description(field.description);
+                                        field_def
+                                    }
+                                    LIST => panic!("Cannot nest more than 2 lists."),
+                                    // [[Type!]]!
+                                    NON_NULL => {
+                                        let type_ref = type_ref.of_type.unwrap();
+                                        match type_ref.kind {
+                                            SCALAR | OBJECT | INTERFACE | UNION | ENUM
+                                            | INPUT_OBJECT => {
+                                                let ty = FieldType::new_type(
+                                                    type_ref.name.unwrap(),
+                                                    false,
+                                                    None,
+                                                );
+                                                let list1 = FieldType::new_list(ty, true);
+                                                let list2 = FieldType::new_list(list1, false);
+                                                let mut field_def = Field::new(field.name, list2);
+                                                field_def.description(field.description);
+                                                field_def
+                                            }
+                                            LIST => panic!("Cannot nest more than 2 lists."),
+                                            ty => panic!("Unknown type: {:?}", ty),
+                                        }
+                                    }
+                                    ty => panic!("Unknown type: {:?}", ty),
+                                }
+                            }
+                            ty => panic!("Unknown type: {:?}", ty),
+                        }
+                    }
+                    ty => panic!("Unknown type: {:?}", ty),
+                }
+            }
+            Other(ty) => panic!("Unknown type: {}", ty),
+        }
+    }
+}
+
+impl TryFrom<Introspection> for Schema {
+    type Error = &'static str;
+
+    fn try_from(src: Introspection) -> Result<Self, Self::Error> {
+        match src.schema {
+            Some(s) => Ok(Self {
+                types: s.types,
+                directives: s.directives,
+            }),
+            None => Err("Schema not found in Introspection Result."),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn it_build_simple_schema() {}
```

Comment:
Couldn't you manually execute introspection queries  and store the results in files?
(With the old `apollo` CLI, you could get those by running something like `apollo schema:download --endpoint=http://localhost:8080/graphql schema.json`.) Or am I misunderstanding the issue?

Reply:
What you're suggesting is not possible as Schema is currently created for an `IntrospectionResult`. `.json` file will not work. A `.json` file would  have worked if `IntrospectionResult` had a `Serialize` implementation, but this implementation doesn't exist and the following code does not work:
```rust
        let file = File::open("src/introspection/introspection.json")
            .expect("File should be able to be opened.");
        let result: IntrospectionResult =
            serde_json::from_reader(file).expect("File is not a proper JSON.");
```

Reply:
I don't understand. Since executing an introspection query against a server would return the same JSON contents, and we are able to load those into an `IntrospectionResult`, why is this different?